File: Day12/descending_order.py
import numbers


def descending_order(number):
    if number < 0:
        raise ValueError('Number cannot be negative')
    for digit in str(number):
        numbers = list(str(number))
        for num in range(len(numbers)):
            for char in range(len(numbers)):
                if numbers[num] > numbers[char]:
                    temp = numbers[num]
                    numbers[num] = numbers[char]
                    numbers[char] = temp
                    word = "".join(numbers)
    return word


numbers = 12345
print(descending_order(numbers))

File: Day12/DescendingOrder.java
import java.util.Arrays;
public class DescendingOrder{

	public static void main(String [] args){
		int number = 194355;
		System.out.print(Descending(number));
	}

	public static String Descending(int number){
		int length = String.valueOf(number).length();
		int [] newNumber = new int [length];
		int count = 0;
		while(number > 0){
			int remainder = number % 10;
			newNumber[count] = remainder;
			number /= 10;
			count++;
		} 
		int [] sortedNumbers = sortArray(newNumber);
		String joined = joinedNumbers(sortedNumbers);
		return joined;

	}


	public static int[] sortArray(int [] numbers){
		for(int count = 0; count < numbers.length; count++){
			for(int counter = count + 1; counter < numbers.length; counter++){
					if(numbers[count] < numbers[counter]){
						int temp = numbers[count];
						numbers[count] = numbers[counter];
						numbers[counter] = temp;
					}
			}
		

		}
		return numbers;

	}

	public static String joinedNumbers(int [] numbers){
		String joined = "";
		for(int count = 0; count < numbers.length; count++){
			joined += numbers[count];

		}
		return joined;

	}



}